pipeline-model-api/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/ast/ModelASTValue.java

```diff
@@ -31,7 +31,7 @@
  * @author Andrew Bayer
  * @author Kohsuke Kawaguchi
  */
-public abstract class ModelASTValue extends ModelASTElement implements ModelASTMethodArg {
+public abstract class ModelASTValue extends ModelASTElement implements ModelASTMethodArg, ModelASTEnvironmentValue {
```

Comment:
It means that every `ModelASTValue` *could* be used as a `ModelASTEnvironmentValue`.